content/en/blog/2021/external-locality-failover/index.md

```diff
@@ -0,0 +1,215 @@
+---
+title: "Configuring failover for external services"
+description: Learn how to configure locality load balancing and failover for endpoints that are outside of your mesh.
+publishdate: 2021-06-04
+attribution: "Ram Vennam (Solo.io)"
+keywords: [locality,region,failover,Istio,outlier,external]
+---
+
+Istio’s powerful APIs can be used to solve a variety of service mesh use cases. Many users know about its strong ingress and east-west capabilities but it also offers many features for egress (outgoing) traffic. This is especially useful when your application needs to talk to an external service - such as a database endpoint provided by a cloud provider. There are often multiple endpoints to chose from depending on where your workload is running. For example, Amazon's DynamoDB provides [several endpoints](https://docs.aws.amazon.com/general/latest/gr/ddb.html) across their regions. You typically want to choose the endpoint closest to your workload for latency reasons, but you may need to configure automatic failover to another endpoint in case things are not working as expected.
+
+Similar to services running inside the service mesh, you can configure Istio to detect outliers and failover to a healthy endpoint, while still being completely transparent to your application. In this example, we’ll use Amazon DynamoDB endpoints and pick a primary region that is the same or close to workloads running in a Google Kubernetes Engine (GKE) cluster. We’ll also configure a failover region.
+
+|Routing|Endpoint|
+|--- |--- |
+|Primary|http://dynamodb.us-east-1.amazonaws.com|
+|Failover|http://dynamodb.us-west-1.amazonaws.com|
+
+![failover](./external-locality-failover.png)
+
+## Define external endpoints using a ServiceEntry
+
+First, determine the location of your workloads.
+
+{{< text bash >}}
+$ kubectl describe node | grep failure-domain.beta.kubernetes.io/region
+                    failure-domain.beta.kubernetes.io/region=us-east1
+                    failure-domain.beta.kubernetes.io/region=us-east1
+{{< /text >}}
+
+In this example, the GKE cluster nodes are running in `us-east1`
+
+Next, apply a `ServiceEntry` to create a `mydb.com` service that’s backed by the two DynamoDB endpoints. Set the `locality` of your primary endpoint to the same region as your workload.
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1beta1
+kind: ServiceEntry
+metadata:
+ name: external-svc-dns
+spec:
+ hosts:
+ - mydb.com
+ location: MESH_EXTERNAL
+ ports:
+ - number: 80
+   name: http
+   protocol: HTTP
+ resolution: DNS
+ endpoints:
+ - address: dynamodb.us-east-1.amazonaws.com
+   locality: us-east1
+   ports:
+     http: 80
+ - address: dynamodb.us-west-1.amazonaws.com
+   locality: us-west
```

Comment:
one more nit: one is us-east*1*, the other is us-west, throughout?

(This might be confusing Google regions with AWS regions)

Reply:
because the cluster nodes are only in `us-east1`, it doesn't matter what you set the locality of the other endpoints to. I didn't set it to `us-west1` because that seemed oddly specific as it doesn't map to any value.

Reply:
`other`? `failover`?